test/test_um2netcdf.py

```diff
@@ -206,124 +212,130 @@ def fake_out_path():
-    """Attempt end-to-end process() test, dropping cubes requiring masking."""
+def test_process_cubes_no_heaviside_drop_cubes(ta_plev_cube, precipitation_flux_cube,
+                                               geo_potential_cube, mule_vars, std_args,
+                                               fake_in_path, fake_out_path):
+    """Attempt end-to-end process_cubes() test, dropping cubes requiring masking."""
+
+    # include cubes requiring both heaviside uv & t cubes to filter, to
+    # ensure both uv/t dependent cubes are dropped
+    cubes = [ta_plev_cube, precipitation_flux_cube, geo_potential_cube]
+
+    # mock fix_time_coord to avoid adding difficult to replicate
+    # iris methods in DummyCube.
     with (
-        # use mocks to prevent mule data extraction file I/O
-        mock.patch("mule.load_umfile"),
-        mock.patch("umpost.um2netcdf.process_mule_vars") as m_mule_vars,
-        mock.patch("iris.load") as m_iris_load,
-        mock.patch("iris.fileformats.netcdf.Saver") as m_saver,  # prevent I/O
-        mock.patch("umpost.um2netcdf.cubewrite"),
+        mock.patch("umpost.um2netcdf.fix_fill_value"),
+        mock.patch("umpost.um2netcdf.fix_time_coord") as m_time_coord,
     ):
-        m_mule_vars.return_value = mule_vars
+        m_time_coord.side_effect = mock_fix_time_no_time_dim
 
-        # include cubes requiring both heaviside uv & t cubes to filter, to
-        # ensure both uv/t dependent cubes are dropped
-        cubes = [ta_plev_cube, precipitation_flux_cube, geo_potential_cube]
-
-        m_iris_load.return_value = cubes
-        m_saver().__enter__ = mock.Mock(name="mock_sman")
         std_args.verbose = True  # test some warning branches
 
         # trying to mask None will break in numpy
         assert precipitation_flux_cube.data is None
 
         # air temp & geo potential should be dropped in process()
         with pytest.warns(RuntimeWarning):
-            processed = um2nc.process(fake_in_path, fake_out_path, std_args)
+            processed = tuple(um2nc.process_cubes(cubes, mule_vars, std_args))
+
+    assert len(processed) == 1
+    cube, _, dim = processed[0]
 
-        assert len(processed) == 1
-        cube = processed[0]
+    assert cube.name() == precipitation_flux_cube.name()
 
-        assert cube is precipitation_flux_cube
+    # contrived testing: if the masking code was reached for some reason,
+    # the test would fail during process()
+    assert cube.data is None  # masking wasn't called/nothing changed
 
-        # contrived testing: if the masking code was reached for some reason,
-        # the test would fail during process()
-        assert cube.data is None  # masking wasn't called/nothing changed
 
+def test_process_cubes_all_cubes_filtered(ta_plev_cube, geo_potential_cube,
+                                          mule_vars, std_args,
+                                          fake_in_path, fake_out_path):
```

Comment:
```suggestion
def test_process_cubes_all_cubes_filtered(ta_plev_cube, geo_potential_cube,
                                          mule_vars, std_args):
```

Removed unused text fixtures